examples/scripts/SPI_MPI.py

```diff
@@ -0,0 +1,101 @@
+import sys
+import os
+import time
+import numpy as np
+import scipy as sp
+import h5py as h5
+from mpi4py import MPI # module required to use MPI
+import argparse
+
+import pysingfel as ps
+import pysingfel.gpu as pg
+from pysingfel.util import asnumpy, xp
+
+ROOT_DIR = "../.."
+if ROOT_DIR not in sys.path:
+    sys.path.append(ROOT_DIR)
+
+def main():
+    # parse user input
+    params = parse_input_arguments(sys.argv)
+    pdb = params['pdb']
+    geom = params['geom']
+    beam = params['beam']
+    numPatterns = int(params['numPatterns'])
+    outDir = params['outDir']
+    saveName = params['saveNameHDF5']
+
+    # set up MPI environment
+    comm = MPI.COMM_WORLD # communication module
+    size = comm.Get_size() # number of processors available
+    rank = comm.Get_rank() # the rank of the process (the rank of a process always ranges from 0 to size-1)
+
+    data = None
+
+    if rank == 0:
+        print ("====================================================================")
+        print ("Running %d parallel MPI processes" % size)
+
+        t_start = MPI.Wtime()
+
+        # load beam
+        beam = ps.Beam(beam)
+
+        # load and initialize the detector
+        det = ps.PnccdDetector(geom=geom, beam=beam)
+
+        # create particle object(s)
+        particle = ps.Particle()
+        particle.read_pdb(pdb, ff='WK')
+
+        data = {"detector": det, "beam": beam, "particle": particle}
+        print ("Broadcasting input to processes...")
+
+    dct = comm.bcast(data,root=0)
+
+    if rank == 0:
+        pattern_shape = det.pedestals.shape # (4, 512, 512)
+
+        f = h5.File(os.path.join(outDir,"SPI_MPI.h5"),"w")
+        dset = f.create_dataset("intensity", shape=(numPatterns,)+pattern_shape, dtype=np.float32, chunks=(1,)+pattern_shape, compression="gzip", compression_opts=4) # (numPatterns, 4, 512, 512)
+
+        print ("Done creating HDF5 file and dataset...")
+
+        n = 0
+        while n < numPatterns:
+            status1 = MPI.Status()
+            result = comm.recv(source=MPI.ANY_SOURCE,status=status1)
```

Comment:
This would be easier to read:
(ind, img) = comm.recv(source=MPI.ANY_SOURCE,status=status1)
and get rid of result[0] and result[1] in lines 69-70